Gordon360/Services/RecIM/ActivityService.cs

```diff
@@ -0,0 +1,103 @@
+                            .Where(l => l.ID == activityID)
+                            .Include(l => l.Team)
+                            .Include(l => l.Series)
+                                .ThenInclude(s => s.Match)
+                            .Include(l => l.ParticipantActivity
+                                .Join(_context.Participant,
+                                    ul => ul.ParticipantID, 
+                                    u => u.ID, 
+                                    (ul,u) => u))
+                            .FirstOrDefault();
+            return result;
+        }
+        public async Task UpdateActivity(Activity updatedActivity)
+        {
+            int activityID = updatedActivity.ID;
+            var activity = _context.Activity
+                            .FirstOrDefault(l => l.ID == updatedActivity.ID);
+            activity.Name = updatedActivity.Name == null ? activity.Name : updatedActivity.Name;
+            activity.Logo = updatedActivity.Logo == null ? activity.Logo : updatedActivity.Logo;
+            activity.RegistrationStart = updatedActivity.RegistrationStart == default 
+                                            ? activity.RegistrationStart 
+                                            : updatedActivity.RegistrationStart;
+            activity.RegistrationEnd = updatedActivity.RegistrationEnd == default
+                                  ? activity.RegistrationEnd
+                                  : updatedActivity.RegistrationEnd;
+            activity.TypeID = updatedActivity.TypeID == default 
+                                ? activity.TypeID 
+                                : updatedActivity.TypeID;
+            activity.SportID = updatedActivity.SportID == default
+                                ? activity.SportID 
+                                : updatedActivity.SportID;
+            activity.StatusID = updatedActivity.StatusID == null
+                                ? activity.StatusID 
+                                : updatedActivity.StatusID;
+            activity.MinCapacity = updatedActivity.MinCapacity == null
+                                   ? activity.MinCapacity
+                                   : updatedActivity.MinCapacity;
+            activity.MaxCapacity = updatedActivity.MaxCapacity == null
+                                   ? activity.MaxCapacity
+                                   : updatedActivity.MaxCapacity;
+            activity.MaxCapacity = updatedActivity.MaxCapacity == null
+                                   ? activity.MaxCapacity
+                                   : updatedActivity.MaxCapacity;
+            activity.SoloRegistration = updatedActivity.SoloRegistration;
+            activity.Completed = updatedActivity.Completed;
+
+            _context.SaveChanges();
+        }
+        public async Task PostActivity(Activity newActivity)
+        {
+            _context.Activity.Add(newActivity);
+            _context.SaveChanges();
```

Comment:
Since this method is async, might as well use the async ef core methods:
```suggestion
            await _context.Activity.AddAsync(newActivity);
            _context.SaveChangesAsync();
```